unit/testing-utils/require_type.cpp

```diff
@@ -78,3 +93,109 @@ code_typet::parametert require_type::require_parameter(
   REQUIRE(param != function_type.parameters().end());
   return *param;
 }
+
+/// Helper function for testing that java_generic_parametert types match
+/// a given expectation.
+///
+/// \param param The generic parameter to test
+/// \param expected The expected value of the parameter
+/// \return true if the generic parameter type meets the expectations
+bool require_java_generic_parametert_expectation(
+  const java_generic_parametert &param,
+  const require_type::expected_type_parametert &expected)
+{
+  switch(expected.kind)
+  {
+    case require_type::type_parameter_kindt::Var:
+      REQUIRE(!is_java_generic_inst_parameter((param)));
+      REQUIRE(param.type_variable().get_identifier()==expected.description);
+      return true;
```

Comment:
Any reason for the return?

Reply:
As used for the algorithm below. It might be better to use `std::foreach` since it isn't really checking equality but more running some requires on each entry (as you said, the `false` is unreachable)

Reply:
As discussed face-to-face, we actually want to do a pair-wise iteration over two vectors, so std::foreach would lead to a more obtuse implementation.